### ISSUE TITLE
Unable to upgrade to v3.3.x due to dependency on AppSync

### ISSUE THREAD

**Issue by HishamAli81** [anomalyco/sst#5156]
I am trying to upgrade from SST v3.2.76 to v3.3.38 and am immediately getting this error when running `sst deploy`:

```bash
✕  aws: operation error AppSync: CreateApi, https response error StatusCode: 0, RequestID: , request send failed, Post "https://appsync.ap-southeast-5.amazonaws.com/v2/apis": dial tcp: lookup appsync.ap-southeast-5.amazonaws.com: no such host
```

As you can tell from the error, I'm trying to deploy to the ap-southeast-5 AWS region which currently doesn't yet support AppSync.

I'm getting this error even though I'm not using the AppSync component and am even getting it with this minimal config:

```typescript
export default $config({
  app(input) {
    return {
      home: 'aws',
      name: 'sst-app',
      region: 'ap-southeast-5',
    };
  },
  async run() {},
});
``` 

Is there a known workaround for this issue, or is there a hard requirement that SST can only be used in regions that support AppSync?

This issue started with v3.3.1 (https://github.com/sst/sst/releases/tag/v3.3.1), and I'm not 100% sure, but likely due to changes related to this commit: https://github.com/sst/sst/commit/4d48393255d72a8f66ac3ee50212a5fe31a75a8c

**Comment by zohaibakber** [comment]
>As you can tell from the error, I'm trying to deploy to the ap-southeast-5 AWS region which currently doesn't yet support AppSync.

sst internally uses appsync event api for live lambda, sst is trying to provision it, but cant as  not available in your region hence the error. Just change to nearest region with app sync event api.

**Comment by HishamAli81** [comment]
Thanks for the quick response, unfortunately switching regions is not a viable option for a couple reasons:

1. Migration to a different region is a non-trivial task
2. Deploying to ap-southeast-5 is a regulatory requirement of the project

I do see now that AppSync was added as a bootstrap dependency (https://sst.dev/docs/state#bootstrap), which then most likely means that we may need to re-evaluate using SST 😞 (which I'm a big proponent of).

I understand that ap-southeast-5 is a newer region, and it would be tough for SST to cater to every single region and it's current limitations. But seeing how AWS plans to continue to expand it's regions (https://aws.amazon.com/about-aws/global-infrastructure/), my only suggestion would be to keep the footprint of the required bootstrap resources to a bare minimum to better accommodate the hopefully temporary quirks of these newer regions. For example, ECR should only be required if the Cluster component is being used, and AppSync should only be required if Live lambda/`sst dev` is being used. The lack of AWS resources that are only used by `sst dev` for example, should not prohibit someone from being to ship using `sst deploy`.

Again, just my 2 cents while not being overly familiar with the SST codebase and how viable these suggestions are. Overall, think you guys are doing a great job and I'm a fan of what you do 💪

**Comment by jayair** [comment]
Yeah we should address this. I think it's been brought up elsewhere. The deploy should not need this.

**Comment by thdxr** [comment]
hey we did change this around and you no longer need appsync for `sst deploy`
however it's still currently used for `sst dev` but we're thinking through alternatives

**Comment by pauljasperdev** [comment]
I've just realized that an AppSync API is always deployed when running `sst dev` even when nothing else is deployed.

I stumpled upon this because there were AppSync cost I could not figure the origin.

Also, the API is not removed when doing `sst remove` which is pretty unfortunate as it produces cost. Combined with the fact that it is not shown in deploy log and can only be removed manullay in console is not good. Or am I missing something here?

**Comment by jayair** [comment]
@pauljaspersahr it's coming from here https://sst.dev/docs/live#how-it-works

**Comment by pauljasperdev** [comment]
I see thanks.

I was ivestigated my cost a bit more and saw suprisingly high AppSync usage. I started setting up a new Nuxt project in a fresh AWS account and within a couple of days in January I had 5,331,880.29 Connection Minutes and 520,070.00 Operations.

Any idea where this might come from? I am at 400k Events for February aleady...

>[!NOTE]
> EDIT:
> I was able to find the problem. I was missing the VPC in my Nuxt object. Made an [issue](https://github.com/sst/sst/issues/5434#issue-2835904627)

**Comment by HeathHopkins** [comment]
> hey we did change this around and you no longer need appsync for `sst deploy` however it's still currently used for `sst dev` but we're thinking through alternatives

:thumbsup:  AppSync is not available in all AWS accounts.  I work in a security-restricted AWS account, and AppSync has been disabled.  SSM Parameter Store plaintext values automatically get encrypted too within 5 minutes of write.  This means we can't deploy due to SST thinking the JSON stored in `/sst/boostrap` is corrupted.
